testplan/runnable/base.py

```diff
@@ -139,6 +140,33 @@ def validate_lines(d: dict) -> bool:
     return True
 
 
+def collate_for_merging(es):
+    """
+    Group report entries into buckets, where synthesized ones in the same
+    bucket containing the previous non-synthesized one.
+    """
+    res = []
+    i, j = 0, 0
+    while i < len(es):
```

Comment:
for i in range(len(es))

Reply:
well i tend to make `i` & `j` look more explicitly, so i have `i = 0` and use `while` instead of `for`